builtin/providers/google/resource_compute_region_instance_group_manager.go

```diff
@@ -0,0 +1,420 @@
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	region, err := getRegion(d, config)
+	if err != nil {
+		return err
+	}
+
+	var targetSize int64 = 1
+	if v, ok := d.GetOk("target_size"); ok {
+		targetSize = int64(v.(int))
+	}
+
+	// Build the parameter
+	manager := &compute.InstanceGroupManager{
+		Name:             d.Get("name").(string),
+		BaseInstanceName: d.Get("base_instance_name").(string),
+		InstanceTemplate: d.Get("instance_template").(string),
+		TargetSize:       targetSize,
+	}
+
+	// Set optional fields
+	if v, ok := d.GetOk("description"); ok {
+		manager.Description = v.(string)
+	}
+
+	if v, ok := d.GetOk("named_port"); ok {
+		manager.NamedPorts = getNamedPorts(v.([]interface{}))
+	}
+
+	if attr := d.Get("target_pools").(*schema.Set); attr.Len() > 0 {
+		var s []string
+		for _, v := range attr.List() {
+			s = append(s, v.(string))
+		}
+		manager.TargetPools = s
+	}
+
+	updateStrategy := d.Get("update_strategy").(string)
+	if !(updateStrategy == "NONE" || updateStrategy == "RESTART") {
+		return fmt.Errorf(`Update strategy must be "NONE" or "RESTART"`)
+	}
+
+	log.Printf("[DEBUG] RegionInstanceGroupManager insert request: %#v", manager)
+
+	op, err := config.clientCompute.RegionInstanceGroupManagers.Insert(
+		project, region, manager).Do()
+	if err != nil {
+		return fmt.Errorf("Error creating RegionInstanceGroupManager: %s", err)
+	}
+
+	d.SetId(manager.Name)
+
+	err = computeOperationWaitRegion(config, op, project, region, "Creating RegionInstanceGroupManager")
+	if err != nil {
+		return err
+	}
+
+	return resourceComputeRegionInstanceGroupManagerRead(d, meta)
+}
+
+func resourceComputeRegionInstanceGroupManagerRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	region, err := getRegion(d, config)
+	if err != nil {
+		return err
+	}
+
+	service, err := config.clientCompute.RegionInstanceGroupManagers.Get(
+		project, region, d.Id()).Do()
+	if err != nil {
+		if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 404 {
+			log.Printf("[WARN] Removing Instance Group Manager %q because it's gone", d.Get("name").(string))
+
+			d.SetId("")
+
+			return nil
+		}
+
+		return fmt.Errorf("Error reading service: %s", err)
+	}
+
+	d.Set("base_instance_name", service.BaseInstanceName)
+	d.Set("instance_template", service.InstanceTemplate)
+	d.Set("name", service.Name)
+	d.Set("description", service.Description)
+	d.Set("project", project)
+	d.Set("target_size", service.TargetSize)
+	d.Set("target_pools", service.TargetPools)
+	d.Set("named_port", flattenNamedPorts(service.NamedPorts))
+	d.Set("fingerprint", service.Fingerprint)
+	d.Set("instance_group", service.InstanceGroup)
+	d.Set("target_size", service.TargetSize)
+	d.Set("self_link", service.SelfLink)
+	d.Set("update_strategy", "RESTART") //this field doesn't match the manager api, set to default value
```

Comment:
This would cause a perpetual diff if the user set update_strategy to anything else, I imagine. Why not try to read it out of `d` and set it to the default only if `d` doesn't have a value for it?